File: test/de/kimminich/kata/tcg/GameIntegrationTest.java
package de.kimminich.kata.tcg;

import de.kimminich.kata.tcg.strategy.AiStrategy;
import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.notNullValue;

public class GameIntegrationTest {

    private Game game;

    @Test
    public void gameWillHaveWinnerWhenOver() {
        Player player1 = new Player("1", new AiStrategy());
        Player player2 = new Player("2", new AiStrategy());
        game = new Game(player1, player2);

        game.run();

        assertThat(game.getWinner(), is(notNullValue()));
    }

}

File: tcg-kotlin/src/test/kotlin/de/kimminich/kata/tcg/GameIntegrationTest.kt
package de.kimminich.kata.tcg

import de.kimminich.kata.tcg.strategy.AiStrategy
import org.hamcrest.CoreMatchers.`is`
import org.hamcrest.CoreMatchers.notNullValue
import org.hamcrest.MatcherAssert
import kotlin.test.Test

class GameIntegrationTest {
    val player1 = Player("1", AiStrategy())
    val player2 = Player("2", AiStrategy())
    val game: Game = Game(player1, player2)
    @Test
    fun gameWillHaveWinnerWhenOver() {
        game.run()
        MatcherAssert.assertThat(game.winner, `is`(notNullValue()))
    }
}
